evaluate/repository.go

```diff
@@ -19,16 +20,63 @@ import (
 	"github.com/symflower/eval-dev-quality/util"
 )
 
+// repositoryConfiguration holds the configuration of a repository.
+type repositoryConfiguration struct {
+	Tasks []task.Identifier
+}
+
+// validate validates the configuration.
+func (rc *repositoryConfiguration) validate() (err error) {
+	if len(rc.Tasks) == 0 {
+		return pkgerrors.Errorf("empty list of tasks in configuration")
+	}
+
+	for _, taskIdentifier := range rc.Tasks {
+		if !task.LookupIdentifier[taskIdentifier] {
+			return pkgerrors.Errorf("task identifier %q unknown", taskIdentifier)
+		}
+	}
+
+	return nil
+}
+
+// defaultConfiguration holds the default configuration object if there exists no configuration file.
+var defaultConfiguration = repositoryConfiguration{
+	Tasks: task.AllIdentifiers,
+}
+
 // Repository holds data about a repository.
 type Repository struct {
+	repositoryConfiguration
+
 	// Name holds the name of the repository.
 	Name string
 	// DataPath holds the path to the repository.
 	DataPath string
 }
 
+// loadConfiguration loads the configuration from the dedicated configuration file.
+func (r *Repository) loadConfiguration() (err error) {
+	configurationFilePath := filepath.Join(r.DataPath, "repository.json")
+
+	data, err := os.ReadFile(configurationFilePath)
+	if errors.Is(err, os.ErrNotExist) {
+		r.repositoryConfiguration = defaultConfiguration
+
+		return nil
+	} else if err != nil {
+		return pkgerrors.Wrap(err, configurationFilePath)
+	}
+
+	if err := json.Unmarshal(data, &r.repositoryConfiguration); err != nil {
+		return pkgerrors.Wrap(err, configurationFilePath)
+	}
+
+	return r.repositoryConfiguration.validate()
+}
+
 // Evaluate evaluates a repository with the given model and language.
-func (r *Repository) Evaluate(logger *log.Logger, resultPath string, model evalmodel.Model, language language.Language) (repositoryAssessment metrics.Assessments, problems []error, err error) {
+func (r *Repository) Evaluate(logger *log.Logger, resultPath string, model evalmodel.Model, language language.Language, taskIdentifier task.Identifier) (repositoryAssessment metrics.Assessments, problems []error, err error) {
 	log, logClose, err := log.WithFile(logger, filepath.Join(resultPath, evalmodel.CleanModelNameForFileSystem(model.ID()), language.ID(), r.Name+".log"))
 	if err != nil {
 		return nil, nil, err
```

Comment:
Please  consider changing the logging to also incorporate the task, and add the task also to the log call below (which I cannot select for review - thanks GitHub!).

Reply:
Should we maybe change the log file location as well? I.e. different files for different tasks, otherwise they will clump up in one log file.

Reply:
true true true